framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3052,11 +3053,23 @@ private AnnotationMirrorSet getExceptionParameterLowerBoundAnnotationsCached() {
      * @param tree a CatchTree to check
      */
     protected void checkExceptionParameter(CatchTree tree) {
-        AnnotationMirrorSet requiredAnnotations =
+        Set<? extends AnnotationMirror> requiredAnnotations =
                 getExceptionParameterLowerBoundAnnotationsCached();
         VariableTree excParamTree = tree.getParameter();
         AnnotatedTypeMirror excParamType = atypeFactory.getAnnotatedType(excParamTree);
-
+        //        List<AnnotatedTypeMirror> exceptionList =
+        // atypeFactory.getAnnotatedType(methodTree).getThrownTypes();
```

Comment:
@wmdietl  I try to compare the type of exception that method body throw and the type of catch parameter. Do you have any suggestion on how to get what exception the method body throws?

Reply:
@wmdietl I think we should allow the following code as the exception may not be catched like how Java handle this:

```
void foo() throw @UnknowInitialization myException {}

void test(){
    try {
        foo()
    } catch (MyException e) {}
}
```